### ISSUE TITLE
spacemacs/report-issue fails if I set any layer variables with periods in them

### ISSUE THREAD

**Issue by dwang20151005** [syl20bnr/spacemacs#5901]
#### Description
When `dotspacemacs-configuration-layers` has any symbols with periods in them (e.g., `(jane :variables Omake\.Server\.program)`), spacemacs/report-issue fails with an error `Invalid use of ``\\' in replacement text`.

For me, this can be fixed by the following patch.  No idea if this is acceptable generally.

    diff --git a/core/core-spacemacs.el b/core/core-spacemacs.el
    index 9feebb2..65bfcc2 100644
    --- a/core/core-spacemacs.el
    +++ b/core/core-spacemacs.el
    @@ -265,7 +265,7 @@ with the message pre-filled with template and information."
         (setq template (replace-regexp-in-string
                         "%SYSTEM_INFO%"
                         (spacemacs//describe-system-info-string)
    -                    template [keep-case]))
    +                    template [keep-case] t))
         ;; Include the backtrace directly in the template, if it exists
         (setq template (replace-regexp-in-string
                         "%BACKTRACE%"
    
#### Reproduction guide
- Start Emacs 
- Put a symbol with a period into your .spacemacs (see my "System info")
- Call spacemacs/report-issue

*Observed behaviour:*
Error described above.

*Expected behaviour:*
Expected a browser window.

#### System Info
- OS: gnu/linux
- Emacs: 24.5.2
- Spacemacs: 0.105.13
- Spacemacs branch: chrome-layer-plain-text-gmail (rev. 107b53f)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
((auto-completion :variables auto-completion-return-key-behavior 'complete auto-completion-tab-key-behavior 'cycle auto-completion-enable-help-tooltip t)
 (chrome :variables chrome-gmail-rich-text-format nil)
 emacs-lisp
 (jane :variables jane-jane-elisp-path "/usr/local/home/dwang/workspaces/jane/emacs/dev/spacemacs/jane-tweaks/rewrite/ share /app/emacs/" Omake\.Server\.program "/j/office/app/emacs/prod/jane-elisp/bin/build-manager")
 markdown org sql
 (version-control :variables version-control-diff-tool 'diff-hl)
 vinegar w3m)
```

#### Backtrace
```
 Debugger entered--Lisp error: (error "Invalid use of `\\' in replacement text")
  replace-match("#### System Info\n- OS: gnu/linux\n- Emacs: 24.5.2\n- Spacemacs: 0.105.13\n- Spacemacs branch: chrome-layer-plain-text-gmail (rev. 107b53f)\n- Graphic display: t\n- Distribution: spacemacs\n- Editing style: vim\n- Completion: helm\n- Layers:\n```elisp\n((auto-completion :variables auto-completion-return-key-behavior 'complete auto-completion-tab-key-behavior 'cycle auto-completion-enable-help-tooltip t)\n (chrome :variables chrome-gmail-rich-text-format nil)\n emacs-lisp\n (jane :variables jane-jane-elisp-path \"/usr/local/home/dwang/workspaces/jane/emacs/dev/spacemacs/jane-tweaks/rewrite/+share+/app/emacs/\" Omake\\.Server\\.program \"/j/office/app/emacs/prod/jane-elisp/bin/build-manager\")\n markdown org sql\n (version-control :variables version-control-diff-tool 'diff-hl)\n vinegar w3m)\n```\n" [keep-case] nil "%SYSTEM_INFO%" nil)
  replace-regexp-in-string("%SYSTEM_INFO%" "#### System Info\n- OS: gnu/linux\n- Emacs: 24.5.2\n- Spacemacs: 0.105.13\n- Spacemacs branch: chrome-layer-plain-text-gmail (rev. 107b53f)\n- Graphic display: t\n- Distribution: spacemacs\n- Editing style: vim\n- Completion: helm\n- Layers:\n```elisp\n((auto-completion :variables auto-completion-return-key-behavior 'complete auto-completion-tab-key-behavior 'cycle auto-completion-enable-help-tooltip t)\n (chrome :variables chrome-gmail-rich-text-format nil)\n emacs-lisp\n (jane :variables jane-jane-elisp-path \"/usr/local/home/dwang/workspaces/jane/emacs/dev/spacemacs/jane-tweaks/rewrite/+share+/app/emacs/\" Omake\\.Server\\.program \"/j/office/app/emacs/prod/jane-elisp/bin/build-manager\")\n markdown org sql\n (version-control :variables version-control-diff-tool 'diff-hl)\n vinegar w3m)\n```\n" "<PLEASE REPLACE ALL UPPERCASE EXPRESSIONS AND DELETE THIS LINE>\n#### Description\nDESCRIPTION OF THE PROBLEM\n\n#### Reproduction guide\n- Start Emacs\n- REPRODUCTION STEPS\n\n*Observed behaviour:*\nDESCRIPTION OF THE OBSERVED BEHAVIOUR\n\n*Expected behaviour:*\nDESCRIPTION OF THE EXPECTED BEHAVIOUR\n\n(%LAST_KEYS%)\n\n%SYSTEM_INFO%\n\n#### Backtrace\n```\n%BACKTRACE%\n```\n" [keep-case])
  (setq template (replace-regexp-in-string "%SYSTEM_INFO%" (spacemacs//describe-system-info-string) template [keep-case]))
  (let* ((url "http://github.com/syl20bnr/spacemacs/issues/new?body=") (template (let ((temp-buffer (generate-new-buffer " *temp*"))) (save-current-buffer (set-buffer temp-buffer) (unwind-protect (progn (insert-file-contents-literally ...) (buffer-string)) (and (buffer-name temp-buffer) (kill-buffer temp-buffer))))))) (setq template (replace-regexp-in-string "%SYSTEM_INFO%" (spacemacs//describe-system-info-string) template [keep-case])) (setq template (replace-regexp-in-string "%BACKTRACE%" (if (get-buffer "*Backtrace*") (save-current-buffer (set-buffer "*Backtrace*") (buffer-substring-no-properties (point-min) (min (point-max) 1000))) "BACKTRACE IF RELEVANT") template [keep-case])) (setq template (replace-regexp-in-string "(%LAST_KEYS%)\n" (if (and arg (y-or-n-p (concat "Do you really want to " "include your last pressed keys? It " "may include some sensitive data."))) (concat (spacemacs//describe-last-keys-string) "\n") "") template [keep-case])) (setq url (url-encode-url (concat url template))) (setq url (replace-regexp-in-string "#" "%23" url)) (browse-url url))
  spacemacs/report-issue(nil)
  call-interactively(spacemacs/report-issue nil nil)
  command-execute(spacemacs/report-issue)

```

**Comment by JAremko** [comment]
I'll summon @StreakyCobra

**Comment by StreakyCobra** [comment]
The solution you proposed is probably fine. You can PR it.

**Comment by dwang20151005** [comment]
The way I got my dev setup was a bit different than was documented on your [beginner's guide](https://github.com/syl20bnr/spacemacs/wiki/Beginner's-Guide-to-Contributing-a-Pull-Request-to-Spacemacs).  I don't know if this is a bad idea for some reason or maybe could be added.

I wanted to use my existing clone of the spacemacs repo, so I did this:

    git remote add dwang git@github.com:dwang20151005/spacemacs.git
    git fetch dwang
    git branch fix-report-issue-for-escaped-dots dwang/develop
    git push --setup-upstream dwang fix-report-issue-for-escaped-dots
    git checkout fix-report-issue-for-escaped-dots

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
